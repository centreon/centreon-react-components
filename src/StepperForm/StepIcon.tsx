import React from 'react';

import { makeStyles, StepIconProps } from '@material-ui/core';
import Avatar from '@material-ui/core/Avatar';
import Check from '@material-ui/icons/Check';

const useStepIconStyles = makeStyles((theme) => ({
  root: {
    color: '#000000',
    display: 'flex',
    height: 22,
    alignItems: 'center',
  },
  avatar: {
    width: 20,
    height: 20,
    color: '#000000',
    backgroundColor: '#ffffff',
    fontSize: '0.8rem',
  },
  avatarActive: {
    color: '#ffffff',
    backgroundColor: theme.palette.primary.main,
    boxShadow: '0 1px 2px 1px rgba(0,0,0,.25)',
  },
  avatarCompleted: {
    backgroundColor: theme.palette.primary.main,
  },
  completed: {
    color: '#ffffff',
    zIndex: 1,
    fontSize: 18,
  },
}));

const StepIcon = ({ active, completed, icon }: StepIconProps): JSX.Element => {
  const classes = useStepIconStyles();

  return (
    <div className={classes.root}>
      {completed ? (
        <Avatar className={`${classes.avatar} ${classes.avatarCompleted}`}>
          <Check className={classes.completed} />
        </Avatar>
      ) : (
        <Avatar
          className={`${classes.avatar} ${active ? classes.avatarActive : ''}`}
        >
          {icon}
        </Avatar>
      )}
    </div>
  );
};

export default StepIcon;
