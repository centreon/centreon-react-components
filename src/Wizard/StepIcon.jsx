import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'clsx';
import { makeStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import Check from '@material-ui/icons/Check';

const useStepIconStyles = makeStyles((theme) => ({
  root: {
    color: '#000000',
    display: 'flex',
    height: 22,
    alignItems: 'center',
  },
  avatar: {
    width: 20,
    height: 20,
    color: '#000000',
    backgroundColor: '#ffffff',
    fontSize: '0.8rem',
  },
  avatarActive: {
    color: '#ffffff',
    backgroundColor: theme.palette.primary.main,
    boxShadow: '0 1px 2px 1px rgba(0,0,0,.25)',
  },
  avatarCompleted: {
    backgroundColor: theme.palette.primary.main,
  },
  completed: {
    color: '#ffffff',
    zIndex: 1,
    fontSize: 18,
  },
}));

const StepIcon = ({ active, completed, icon }) => {
  const classes = useStepIconStyles();

  return (
    <div className={classes.root}>
      {completed ? (
        <Avatar className={`${classes.avatar} ${classes.avatarCompleted}`}>
          <Check className={classes.completed} />
        </Avatar>
      ) : (
        <Avatar
          className={classNames(classes.avatar, {
            [classes.avatarActive]: active,
          })}
        >
          {icon}
        </Avatar>
      )}
    </div>
  );
};

StepIcon.propTypes = {
  active: PropTypes.bool,
  completed: PropTypes.bool,
  icon: PropTypes.node,
};

StepIcon.defaultProps = {
  active: false,
  completed: false,
  icon: null,
};

export default StepIcon;
