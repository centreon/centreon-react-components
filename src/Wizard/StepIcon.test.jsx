import React from 'react';
import { render } from '@testing-library/react';
import StepIcon from './StepIcon';

describe('StepIcon', () => {
  it('renders completed step correctly', () => {
    const { container } = render(<StepIcon completed />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders active step correctly', () => {
    const { container } = render(<StepIcon active icon={1} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders pending step correctly', () => {
    const { container } = render(<StepIcon icon={2} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
