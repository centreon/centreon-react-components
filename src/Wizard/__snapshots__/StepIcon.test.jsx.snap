// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepIcon renders active step correctly 1`] = `
<div
  class="makeStyles-root-22"
>
  <div
    class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-23 makeStyles-avatarActive-24 MuiAvatar-colorDefault"
  >
    1
  </div>
</div>
`;

exports[`StepIcon renders completed step correctly 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 makeStyles-avatarCompleted-4 MuiAvatar-colorDefault"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-completed-5"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
      />
    </svg>
  </div>
</div>
`;

exports[`StepIcon renders pending step correctly 1`] = `
<div
  class="makeStyles-root-34"
>
  <div
    class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-35 MuiAvatar-colorDefault"
  >
    2
  </div>
</div>
`;
